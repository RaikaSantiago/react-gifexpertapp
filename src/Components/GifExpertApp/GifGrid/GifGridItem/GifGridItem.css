.card {
    align-content: center;
    border: 1px solid gray;
    border-radius: 6px;
    margin-bottom: 10px;
    margin-right: 10px;
    overflow: hidden;
    text-align: center;
    width: 300px;
    height: 300px;
}

.h4-back{
    background-color: black;
    color: white;
    margin-top: 0px;
    margin-bottom: 0px;
}