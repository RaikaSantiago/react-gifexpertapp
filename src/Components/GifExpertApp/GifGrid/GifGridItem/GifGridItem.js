import React from 'react';
import './GifGridItem.css';
import PropTypes from 'prop-types';
export const GifGridItem = ({url,title}) => {

  return (
    <>
      <div className='card animate__animated animate__flash'>
        <h4 className='h4-back'>{title}</h4>
        <img src={url} alt={title} height="100%" width="100%" />
      </div>
    </>
    
  )
}

GifGridItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}
