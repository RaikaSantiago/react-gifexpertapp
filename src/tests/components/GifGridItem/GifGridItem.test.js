import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../../Components/GifExpertApp/GifGrid/GifGridItem/GifGridItem';

describe('Pruebas <GifGridItem  />', () => { 

    const title = 'Titulo';
    const url = 'https://localhost/prueba.jpg'

    let wrapper = shallow( <GifGridItem title={title}  url={url} />);
    beforeEach( () => {
        wrapper = shallow( <GifGridItem title={title}  url={url} />);
    })

    test('Debe de mostrar <GifGridItem  /> correctamente', () => { 

        expect( wrapper).toMatchSnapshot();
    })

    test('Debe de tener un <h4></h4>  con el titulo', () => { 
        const h4 = wrapper.find('h4');
        expect(h4.text().trim()).toBe(title);
    })

    test('Debe de tener la <img> igual al url y alt  de los props', () => { 
        const img = wrapper.find('img');
        // expect(img.props().src).toBe(url);
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Debe tener la clase animate__animated animate__flash', () => { 
        const div = wrapper.find('div');

        console.log(div.html());
    })

})