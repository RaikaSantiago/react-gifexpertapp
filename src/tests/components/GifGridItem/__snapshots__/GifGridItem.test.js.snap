// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifGridItem  /> Debe de mostrar <GifGridItem  /> correctamente 1`] = `
<Fragment>
  <div
    className="card animate__animated animate__flash"
  >
    <h4
      className="h4-back"
    >
      Titulo
    </h4>
    <img
      alt="Titulo"
      height="100%"
      src="https://localhost/prueba.jpg"
      width="100%"
    />
  </div>
</Fragment>
`;
